### PR TITLE
[tests-only][full-ci]Add `/Shares` realated tests from core

### PR DESCRIPTION
**Pull request by SagarGi**
### Description

This PR adds the tests for checksum related to `/Shares` which is removed from this PR https://github.com/owncloud/core/pull/40271 as `/Shares` related implementation is not in core and is shifted to ocis.

### Related issue
https://github.com/owncloud/ocis/issues/4154#issuecomment-1206026045

Note: Most of the tests for the checksum is run from core on ocis. This test added on ocis is only related to `/Shares` (Shares Jail)

Note: Expected to failure for this PR is removed in this PR https://github.com/owncloud/ocis/pull/4364

### CODE REVIEW

File: tests/acceptance/features/apiSpaces/checksum.feature [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  Background:
+    Given user "Alice" has been created with default attributes and without skeleton files
+
+  @files_sharing-app-required @issue-ocis-reva-196
+  Scenario: Sharing a file with checksum should return the checksum in the propfind using new DAV path
+    Given the administrator has set the default folder for received shares to "Shares"
+    And auto-accept shares has been disabled
+    And using spaces DAV path
+    And user "Brian" has been created with default attributes and without skeleton files
```
Comment (SwikritiT) [inline tests/acceptance/features/apiSpaces/checksum.feature:12]: These can go to the background?

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 2]
```diff
@@ -2512,6 +2510,74 @@ public function downloadFile(
 		);
 	}
 
+	/**
+	 * @When /^user "([^"]*)" requests the checksum of "([^"]*)" in space "([^"]*)" via propfind$/
```
Comment (kiranparajuli589) [inline tests/acceptance/features/bootstrap/SpacesContext.php:2514]:
```suggestion
	 * @When /^user "([^"]*)" requests the checksum of "([^"]*)" in space "([^"]*)" using the WebDAV API$/
```

**Review by kiranparajuli589 — COMMENTED** [review]
lgtm :+1:

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 3]
```diff
@@ -2559,6 +2559,74 @@ public function downloadFile(
 		);
 	}
 
+	/**
+	 * @When /^user "([^"]*)" requests the checksum of "([^"]*)" in space "([^"]*)" via propfind using the WebDAV API$/
```
Comment (kiranparajuli589) [inline tests/acceptance/features/bootstrap/SpacesContext.php:2563]:
```suggestion
	 * @When /^user "([^"]*)" requests the checksum of (?:file|folder|entry) "([^"]*)" in space "([^"]*)" via propfind using the WebDAV API$/
```

Comment (kiranparajuli589) [reply tests/acceptance/features/bootstrap/SpacesContext.php:2563]: one leftover

**Review by phil-davis — APPROVED** [review]

**Pull request merged**